pandas/core/tools/numeric.py

```diff
@@ -234,7 +234,8 @@ def to_numeric(
                 set(),
                 coerce_numeric=coerce_numeric,
                 convert_to_masked_nullable=dtype_backend is not lib.no_default
-                or isinstance(values_dtype, StringDtype),
+                or isinstance(values_dtype, StringDtype)
```

Comment:
This is not an issue for the `pyarrow` storage either?

Reply:
No, we want nullable dtype in this case, the mapping is as follows:

- storage=pyarrow_numpy -> NumPy backed
- storage=payrrow -> Nullable
- ArrowDtype string -> PyArrow backed numerics